lib/ai/tools/shell/utils/pty-output.ts

```
@@ -0,0 +1,35 @@
+const START_MARKER_RE = /^.*__START_[a-f0-9]+__.*$/gm;
+
+export const stripSentinelNoise = (text: string): string => {
+  let cleaned = text;
+  cleaned = cleaned.replace(SENTINEL_LINE_RE, "");
+  cleaned = cleaned.replace(ECHO_SENTINEL_RE, "");
+  cleaned = cleaned.replace(ECHO_START_RE, "");
+  cleaned = cleaned.replace(START_MARKER_RE, "");
+
+  const lines = cleaned.split("\n");
+  const filteredLines = lines.filter((line) => {
+    const stripped = line.replace(/\x1b\[[0-9;]*m/g, "").trim();
```

Biome (2.3.14)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)